src/client/src/ui/analytics/views/pnlBar.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import './analytics.scss';
+import numeral from 'numeral';
+
+const NUMERAL_FORMAT = '0,000,000';
+
+export default class PNLBar extends React.Component{
+
+  static propTypes = {
+    index : React.PropTypes.number,
+    model: React.PropTypes.object,
+    isPnL: React.PropTypes.bool,
+    maxMinValues: React.PropTypes.object,
+    ratio: React.PropTypes.ratio,
+    containerWidth: React.PropTypes.number
+  }
+  
+  render(){
+    let baseValue = this.props.isPnL ? this.props.model.basePnl : this.props.model.baseTradedAmount;
+    let isPositive = baseValue > 0;
+    let displayValue = Math.abs(baseValue * this.props.ratio);
+    let xPos = isPositive ? this.props.containerWidth/2 : (this.props.containerWidth/2 - displayValue);
+    let clName = isPositive ? 'indicator green' : 'indicator red';
+    let amountStr = this.props.isPnL ? this.props.model.basePnl : this.props.model.baseTradedAmount;
```

Comment:
Not great practise.  We call a variable a string, but it's a number and then we format it.